### DISCUSSION TITLE
Icaros with non admin account?

### DISCUSSION THREAD

**Discussion by Dave-next** [Xanashi/Icaros#43] [General]
Hi all,
First off thanks to Xanashi and the team for developing ICaros.

Our small video team (3) wants to use it to help their day to day lives but is there anyway to use it without admin rights and registry access?

Security wont be willing to make an exception for this so its pretty much a show stopper for them.

thank in advance and thanks for your work on ICaros!

Dave Kaye

**Comment by Xanashi** [comment]
Hi Dave,
Honestly I'm not entirely sure if it's possible. 

Icaros accesses several reigstry keys in the HKEY_CLASSES_ROOT and HKEY_LOCAL_MACHINE hives, which both requires administrative privileges to modify. 

With that being said, the HKEY_CURRENT_USER registry hive, does have its own "mirror"/version of all these keys, so in theory, 
it would be possible to use these keys to register the shell extensions. 
I'm not sure if Windows would accept these keys as valid alternatives to the HKCR and HKLM keys, when it looks up acceptable shell extensions. 

The Icaros Shell Extensions themselves does not need admin privileges to run though.

Unfortunately, the entire Icaros registration process is coded for users with admin privileges, so I can't make some quick changes to test this. Most of the registration code would have to be rewritten. 

The best way to test it, would be to make a snapshot of the registry on a computer with admin privileges, before registering Icaros, 
and then another snapshot after registering Icaros. Then export the registry changes that happened between the two snapshots, and finally manually change all the HKCR and HKLM keys to HKCU alternatives. Then try and apply this reg file to you non-admin account, and see if Icaros runs. 

Not the most simple solution unfortunately 😅

**Reply by Dave-next** [reply]
thanks for replying!
i'll checkout the registry but i dont want to raise the users hopes

**Comment by AlreadyFree** [comment]
I've been using Icaros previously on Windows 7 and just updated Windows 10. I have my system set up where I have an administrator account but I use a non-administer account for normal operations. So I have access to administer credentials if asked but even so Icaros doesn't seem to have any effect in the non-admin account. It works fine in the admin account but making any changes in the Icaros config panel has no effect when logged in to the non-admin account. Strangely when I first installed icaros on the non-admin account and activated it it would load embedded cover art thumbnails for roughly half of my video files but not all. File properties don't work at all. I've tried building the database in the admin account and saving it in the same folder that is accessible from both accounts but the non-admin account still won't use the database. All the video files are stored an a secondary drive that is fully accessible from both accounts with read/write privileges. Is there any way to make this work?

**Reply by Xanashi** [reply]
I've replied in your issue #99
